### ISSUE TITLE
`min_ident_chars` triggers on imported identifiers, even from the standard library

### ISSUE THREAD

**Issue by 8573** [rust-lang/rust-clippy#12232]
### Summary

`min_ident_chars` triggers on imported identifiers, even from the standard library, even though the user may be unable to do anything about it, especially in the case of the standard library.

The lint's documentation says:

> ### What it does
> Checks for idents which comprise of a single letter.
>
> Note: This lint can be very noisy when enabled; it may be desirable to only enable it temporarily.
>
> ### Why is this bad?
> In many cases it’s not, but at times it can severely hinder readability. Some codebases may wish to disallow this to improve readability.

If the reason for the lint is for an individual codebase to improve its readability, I don't think it makes enough sense for it to complain about identifiers from outside the codebase.

(Also, "comprise of" should be either "comprise" or "consist of".)

### Lint Name

`min_ident_chars`

### Reproducer

I tried this code:

```rust
use core::f64::consts::E;
```

I saw this happen:

```
warning: this ident consists of a single char
    --> src/[...]
     |
[...]|     use core::f64::consts::E;
     |                            ^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#min_ident_chars
```

I expected to see this happen:

No warning about an identifier in the standard library, which I don't control

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03) (built from a source tarball)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 16.0.6
```

**Comment by Centri3** [comment]
You can
```rust
use core::f64::consts::E as EEEEEEEE;
```
Should definitely ignore non local def-ids tho
